src/Tracing/TransactionSource.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Tracing;
+
+/**
+ * This enum represents all the possible types of transaction sources.
+ *
+ * @see https://develop.sentry.dev/sdk/event-payloads/transaction/#transaction-annotations
+ */
+final class TransactionSource implements \Stringable
```

Comment:
It's a mistake that the other classes are missing those protections: if it's a singleton, then it makes no sense to allow it to be cloned nor serialized as you may have back a different instance object.

Reply:
We can follow up on this at a later point. This is nothing critical.

Reply:
It's a bug, definitely. I don't mind if we don't fix the other classes here as it would be out of scope, but the singleton must be implemented properly for the class we're introducing here at least.

Reply:
We didn't run into this yet, so it's not a bug. Please create an issue so we can tackle this at a later stage.